Mathlib/NumberTheory/MaricaSchoenheim.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Combinatorics.SetFamily.FourFunctions
+import Mathlib.Data.Nat.Squarefree
+
+/-!
+# The Marica-Schönheim special case of Graham's conjecture
+
+Graham's conjecture states that if $0 < a_1 < \dots a_n$ are integers, then
+$\max_{i, j} \frac{a_i}{\gcd(a_i, a_j)} \ge n$. This file proves the conjecture when the $a_i$ are
+squarefree as a corollary of the Marica-Schönheim inequality.
+
+## References
+
+[*Applications of the FKG Inequality and Its Relatives*, Graham][Graham1983]
+-/
+
+open Finset
+open scoped BigOperators FinsetFamily
+
+namespace Nat
+
+/-- Statement of Graham's conjecture (which is now a theorem in the literature).
+
+Graham's conjecture states that if $0 < a_1 < \dots a_n$ are integers, then
+$\max_{i, j} \frac{a_i}{\gcd(a_i, a_j)} \ge n$. -/
+def GrahamConjecture (n : ℕ) (f : ℕ → ℕ) : Prop :=
+  ∃ i < n, ∃ j < n, (f i).gcd (f j) * n ≤ f i
```

Comment:
I would prefer `GrahamConjecture` to be true, i.e., include assumptions on `n` and `f`.